### ISSUE TITLE
Remove CsQuery and use AngleSharp instead

### ISSUE THREAD

**Issue by ngosang** [Jackett/Jackett#7292]
Currently we have two libraries to parse HTML: CsQuery and AngleSharp
There are several reasons to remove CsQuery:
* CsQuery is not maintained anymore => https://github.com/jamietre/CsQuery#not-actively-maintained
* AngleSharp performance is better => https://anglesharp.github.io/docs/Performance.html
* We have notice CsQuery requires more memory than AngleSharp in Jackett. => https://github.com/Jackett/Jackett/pull/7094#issuecomment-589938887 https://github.com/Jackett/Jackett/pull/7094#issuecomment-589948133
* There is no point in using two libraries for the same thing

We have 39 indexers using CsQuery. Use this task list to track the progress.
- [x] Abnormal
- [x] Abstract/AvistazTracker
- [x] AnimeTorrents
- [x] BakaBT
- [x] BB
- [x] BitCityReloaded
- [x] BitHdtv
- [x] DigitalHive
- [x] FileList
- [x] FunFile
- [x] Fuzer
- [x] GimmePeers
- [x] HDOlimpo
- [x] HDSpace
- [x] HDTorrents
- [x] Hebits
- [x] ImmortalSeed
- [x] IPTorrents
- [x] LostFilm
- [x] MoreThanTV
- [x] myAmity
- [x] NCore
- [x] NewRealWorld
- [x] Norbits
- [x] Nordicbits
- [x] Partis
- [x] PirateTheNet
- [x] Pretome
- [x] RevolutionTT
- [x] SceneTime
- [x] Shazbat
- [x] SpeedCD
- [x] TorrentBytes
- [x] TorrentDay
- [x] TorrentHeaven
- [x] TorrentLeech
- [x] TorrentSyndikat
- [x] TVstore
- [x] XSpeeds

@Jackett/maintainers What do you think? Could you help with the changes?

@junglebus Thank you for noticing

**Comment by junglebus** [comment]
Sounds good to me
I'll knock over these six since I can test them
IPTorrents
MoreThanTv
AvistazTracker
TorrentDay
TorrentLeech
XSpeeds

**Comment by ngosang** [comment]
@garfield69 I would like to help but I don't have accounts. Could you tell me what trackers you can tests. I will open the PRs blind so you can test.

**Comment by garfield69** [comment]
lostfilm
myamity
norbits
nordicbits
scenetime
speedcd
tvstore

**Comment by cadatoiva** [comment]
I have HDSpace that I can test.

**Comment by cadatoiva** [comment]
On testing HD-Space I found that there's almost a 1-to-1 conversion between csQuery and AngleSharp. I'm making several pull requests with these unoptimized conversions that do nothing special aside from migrating the tracker. In the future, it would make sense to refactor these to be more efficient within the new framework. But these drop in replacements should at least get us off csQuery for now.

**Comment by ngosang** [comment]
@cadatoiva Are you thinking of doing the rest? Do you need help?

**Comment by cadatoiva** [comment]
I'm going to do the rest today and tomorrow as I have time. I stopped to make sure what I was doing would work for the team. I've built a couple of Regex Replace strings for VS so it should make the rest faster. I've gotta run an errand for now, but I'll be back at it in about an hour.

**Comment by cadatoiva** [comment]
That's about half-way done. I'm stopping until tomorrow, if anyone wants to pick up and continue. It's really tedious, even with regex replace

**Comment by ngosang** [comment]
Only 5 left.

**Comment by cadatoiva** [comment]
Checking the boxes if a PR has been made for them, so I can track what's left.

**Comment by cadatoiva** [comment]
NordicBits is a mess that needs to be completely redone and I can't do it blind. Same with Abnormal and Norbits. You can tell that these were written by JS programmers who know jQuery well, which explains why we started with csQuery. However, global use of a `CQ _fDom` variable instead of passing parameters and other *Query "tricks" makes it really hard to follow what's going on.

I'll take care of Shazbat and Xspeeds for now

**Comment by ngosang** [comment]
I think everything is done. When all PRs are merged, we should create a PR to eliminate the CsQuery dependency.

**Comment by cadatoiva** [comment]
Agreed. ReSharper has an `Optimize References` feature that will quickly double check that there are no more dependencies to CsQuery and suggest removing it from the project, I'll run that once all PRs are merged

**Comment by cadatoiva** [comment]
Just need the big 3 merged and we can test again. According to ReSharper, those three trackers make 511 function calls to the CsQuery NuGet package.
